src/iotjs_module_buffer.cpp

```diff
@@ -216,6 +235,10 @@ JHANDLER_FUNCTION(Write) {
 
   BufferWrap* buffer_wrap = BufferWrap::FromJBufferBuiltin(*jbuiltin);
 
+  offset = boundRange(offset, 0, buffer_wrap->length());
```

Comment:
Yes, my_String[4] is out of range. But if I understand your concern correctly, it is *intended* out-of-range.
For example, if we have code like below,
```
var buff = new Buffer("abcd");
buff._builtin.write("ef", 100, 2);
```
The offset `100` is invalid, and nothing should be written in `buff`.
It is accomplished by passing out-of-range offset `4` into below copy function. (we won't enter into the for loop - https://github.com/Samsung/iotjs/blob/master/src/iotjs_module_buffer.cpp#L111)

I think every use of `BoundRange` have this kind of boundary condition.